### PR TITLE
[RFR] Replace tslint with eslint

### PR DESCRIPTION
**Pull request by djhi**
- [x] Replace tslint with eslint
- [x] Adjust configuration
- [x] Replace all tslint comments with eslint ones

### CODE REVIEW

File: packages/ra-core/src/controller/EditController.tsx [hunk 1]
```diff
@@ -133,9 +130,13 @@ const EditController = (props: Props) => {
                 dispatch(updateAction);
             }
         },
-        [id, resource, basePath, record]
+        [resource, id, record, basePath, undoable, dispatch]
     );
 
+    if (!children) {
```
Comment (djhi) [inline packages/ra-core/src/controller/EditController.tsx]: Not really sure about this one but the lint rule for hook complained about us applying hooks conditionnaly

File: packages/ra-core/src/controller/EditController.tsx [hunk 2]
```diff
@@ -104,7 +101,7 @@ const EditController = (props: Props) => {
 
     useEffect(() => {
         dispatch(resetForm(REDUX_FORM_NAME));
-    }, [resource, id, version]);
+    }, [resource, id, version, dispatch]);
```
Comment (djhi) [inline packages/ra-core/src/controller/EditController.tsx]: Not sure about this either. It was adding by "eslint --fix" call

File: examples/simple/src/comments/PostPreview.js [hunk 3]
```diff
@@ -11,7 +11,6 @@ const PostPreview = props => {
         [props.resource, props.id]
     );
     const version = useSelector(state => state.admin.ui.viewVersion);
-    const isLoading = useSelector(state => state.admin.loading > 0);
```
Comment (fzaninotto) [inline examples/simple/src/comments/PostPreview.js]: Why did you remove this one? It forces a redraw when isLoading changes.

File: packages/ra-core/src/controller/EditController.tsx (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-core/src/controller/EditController.tsx]: I'd say ignore the lint rule on that one

File: packages/ra-core/src/controller/ShowController.tsx [hunk 4]
```diff
@@ -73,9 +73,6 @@ interface Props {
 const ShowController = (props: Props) => {
     useCheckMinimumRequiredProps('Show', ['basePath', 'resource'], props);
     const { basePath, children, id, resource } = props;
-    if (!children) {
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/ShowController.tsx]: same

File: packages/ra-core/src/fetch/useQueryWithStore.ts [hunk 5]
```diff
@@ -136,7 +138,7 @@ const useQueryWithStore = (
                     loaded: false,
                 });
             });
-    }, [JSON.stringify({ query, options })]); // deep equality, see https://github.com/facebook/react/issues/14476#issuecomment-471199055
```
Comment (fzaninotto) [inline packages/ra-core/src/fetch/useQueryWithStore.ts]: no, we need that JSON.stringify, it does a deep equal with low overhead

File: packages/ra-ui-materialui/src/list/DatagridLoading.js [hunk 6]
```diff
@@ -72,10 +72,10 @@ export default ({
                             className={classes.expandIconCell}
                         >
                             <IconButton
+                                // TODO: add an aria-label
                                 className={classes.expandIcon}
                                 component="div"
                                 aria-hidden="true"
-                                role="expand"
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/DatagridLoading.js]: why?

File: packages/ra-ui-materialui/src/list/DatagridRow.js [hunk 7]
```diff
@@ -177,9 +177,10 @@ export class DatagridRow extends Component {
                         ) : null
                     )}
                 </TableRow>
-                {expand && expanded && (
+                {// TODO: add proper aria attributes (aria-expanded)
+                expand && expanded && (
                     <TableRow key={`${id}-expand`}>
-                        <TableCell colSpan={colSpan} role="expand-content">
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/DatagridRow.js]: why ?

File: examples/simple/src/comments/PostPreview.js (hunk 3 above)
Comment (djhi) [reply examples/simple/src/comments/PostPreview.js]: Right, it complained about the unused variable

File: packages/ra-ui-materialui/src/list/DatagridLoading.js (hunk 6 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/list/DatagridLoading.js]: Because after looking for it, it appears to not be an existing aria role.

lint rule: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/aria-role.md
list of valid roles: https://www.w3.org/TR/wai-aria/#role_definitions

File: packages/ra-ui-materialui/src/list/DatagridRow.js (hunk 7 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/list/DatagridRow.js]: Same reason. However there are other ways but it deserves a dedicated PR. I'll open an issue to track it

File: packages/ra-core/src/controller/EditController.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-core/src/controller/EditController.tsx]: Why are we doing this though? Shouldn't we throw an error about it ?

Comment (djhi) [reply packages/ra-core/src/controller/EditController.tsx]: Ok but we are breaking the hooks rules here

File: packages/ra-ui-materialui/src/list/DatagridLoading.js (hunk 6 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/list/DatagridLoading.js]: Besides, this button does nothing. It's juste here for placeholder UI while loading

File: packages/ra-ui-materialui/src/list/DatagridRow.js (hunk 7 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/list/DatagridRow.js]: Fixed it at it was needed for the tests

File: packages/ra-core/src/controller/EditController.tsx (hunk 1 above)
Comment (djhi) [reply packages/ra-core/src/controller/EditController.tsx]: Btw, couldn't we fallback to the guesser when we detect there are no children ?

Comment (fzaninotto) [reply packages/ra-core/src/controller/EditController.tsx]: I agree, this is not a normal use case, we should throw. And mark the `children` prop as required.

Also, we can't fall back to an `EditGuesser` because there is no such thing in `ra-core` (it's in `ra-ui-material-ui`)

File: packages/ra-ui-materialui/src/list/DatagridLoading.js (hunk 6 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/list/DatagridLoading.js]: gotcha

File: packages/ra-data-graphcool/src/buildVariables.js [hunk 8]
```diff
@@ -166,10 +165,11 @@ export default introspectionResults => (
                 queryType
             );
         }
-
         case DELETE:
             return {
                 id: params.id,
             };
+        default:
+            console.log('Unknown fetch verb');
```
Comment (fzaninotto) [inline packages/ra-data-graphcool/src/buildVariables.js]: No, unknown fetch verbs are allowed and can be implemented in custom dataProviders

File: packages/ra-data-graphql-simple/src/buildVariables.js [hunk 9]
```diff
@@ -275,10 +270,11 @@ export default introspectionResults => (
                 queryType
             );
         }
-
         case DELETE:
             return {
                 id: preparedParams.id,
             };
+        default:
+            console.log('Unknown fetch verb');
```
Comment (fzaninotto) [inline packages/ra-data-graphql-simple/src/buildVariables.js]: Same, we don't mind unknown fetch verbs

File: packages/ra-data-graphcool/src/buildVariables.js (hunk 8 above)
Comment (djhi) [reply packages/ra-data-graphcool/src/buildVariables.js]: Yes, but it is a good practice to handle the default case right ?

Comment (djhi) [reply packages/ra-data-graphcool/src/buildVariables.js]: Those dataProviders did not handle them at all and returned undefined. How should we handle that ?
